lib/mongo/collection/view/readable.rb

```diff
@@ -133,15 +133,15 @@ def count(opts = {})
           cmd[:limit] = opts[:limit] if opts[:limit]
           cmd[:maxTimeMS] = opts[:max_time_ms] if opts[:max_time_ms]
           cmd[:readConcern] = collection.read_concern if collection.read_concern
-          preference = ServerSelector.get(opts[:read] || read)
+          read_pref = opts[:read] || read_preference
           read_with_retry do
-            server = preference.select_server(cluster, false)
+            server = ServerSelector.get(read_pref || server_selector).select_server(cluster, false)
```

Comment:
Would it still make sense to call`ServerSelector#get` above the retry loop (and store the result in a separate variable from `read_pref`) here and below in `distinct`? I'm not sure how the block is used internally in read_with_retry, but I'm guessing it should be fine to use the same variable in each iteration.